ipv8/test/REST/rtest/peer_interaction_test_env.py

```diff
@@ -0,0 +1,544 @@
+        self._master_peer.stop()
+
+
+class RequestTest(SingleServerSetup):
+
+    """
+    The starting port of the 'other_peer' peers in the test suite
+    """
+    other_peer_port = 7868
+
+    def __init__(self, method_name='runTest'):
+        super(RequestTest, self).__init__(method_name)
+        self.other_peer = None
+        self.default_test_folder = '_trial_temp'
+
+    def setUp(self):
+        super(RequestTest, self).setUp()
+        RequestTest.other_peer_port += 1
+
+    def tearDown(self):
+        super(RequestTest, self).tearDown()
+        RequestTest.gracefully_terminate_peers(self.other_peer)
+
+        if os.path.isdir(self.default_test_folder):
+            rmtree(self.default_test_folder)
+
+    @inlineCallbacks
+    def wait_for_peers(self, dict_param, excluded_peer_mids=None):
+        """
+        Wait until this peer receives a non-empty list of fellow peers in the network
+
+        :param dict_param: the required parameters by the GET request generator for the peers request type
+        :param excluded_peer_mids: A list of peer mids which should not be taken into consideration peers
+        :return: a list of currently known peers in the network
+        """
+        assert isinstance(excluded_peer_mids, (list, set)) or not excluded_peer_mids, "excluded_peer_mids " \
+                                                                                      "must be a list or set or None"
+
+        # Make sure excluded_peer_mids is a set
+        if not excluded_peer_mids:
+            excluded_peer_mids = set()
+        elif isinstance(excluded_peer_mids, list):
+            excluded_peer_mids = set(excluded_peer_mids)
+
+        peer_list = yield self._get_style_requests.make_peers(dict_param)
+        peer_list = set(ast.literal_eval(peer_list))
+
+        # Keep iterating until peer_list is non-empty
+        while not peer_list - excluded_peer_mids:
+            # Wait for 4 seconds before trying again
+            yield deferLater(reactor, 4, lambda: None)
+
+            # Forward and wait for the response
+            peer_list = yield self._get_style_requests.make_peers(dict_param)
+            peer_list = set(ast.literal_eval(peer_list))
+
+        # Return the peer list
+        returnValue(list(peer_list - excluded_peer_mids))
+
+    @inlineCallbacks
+    def wait_for_outstanding_requests(self, dict_param):
+        """
+        Wait until this peer receives a non-empty list of outstanding attestation requests
+
+        :param dict_param: the required parameters by the GET request generator for the outstanding request type
+        :return: a list of outstanding attestation requests
+        """
+        outstanding_requests = yield self._get_style_requests.make_outstanding(dict_param)
+
+        # Keep iterating until peer_list is non-empty
+        while outstanding_requests == "[]":
+            # Wait for 4 seconds before trying again
+            yield deferLater(reactor, 4, lambda: None)
```

Comment:
As this all happens on the same host anyway, you could just use a `0.1` second delay instead.